crypto.go

```diff
@@ -34,37 +48,56 @@ func Equals(sk, other []byte) bool {
 
 // Verify checks the given signature and returns true if it is valid.
 func Verify(pk, msg, signature []byte) bool {
+	pub, err := secpEc.NewPublicKey(pk)
+	if err != nil {
+		return false
+	}
+
+	// Check if the signature is in [R | S | V] style.
 	if len(signature) == 65 {
```

Comment:
Hm. We actually need to _reject_ this signature if it's not exactly 65 bytes long. See https://github.com/ipsn/go-secp256k1/blob/9d62b9f0bc52d16160f79bfb84b2bbf0f6276b03/secp256.go#L160-L162

Reply:
And we still need to assert that `sig[64] < 4` because we're _not_ verifying it.

Reply:
Wait, nvm. I hate all this code.

Reply:
(not your code)

Reply:
See my comment here about `[ParseCompactRecoverableSignature](https://github.com/filecoin-project/go-crypto/pull/5#discussion_r1658070534)` - the underlying lib makes these assertions

Reply:
We'll be covered in `Verify` because we'll just call ECRecover and check the pubKeys match